Python/ibmcloudsql/SQLQuery.py

```diff
@@ -471,6 +1007,97 @@ def get_jobs(self):
             print("Job list retrieval failed with http code {}".format(response.status_code))
         return job_list_df
 
+    def get_jobs_matching(self, job_id_list, status="completed"):
+        """ return the list of job_id that have the given status
+
+        Parameters
+        ----------
+        job_id_list: list
+            List of job_id
+        status : str, optional
+            "completed", or "failed"
+        """
+        results = []
+        for job_id in job_id_list:
+            response = self.get_job(job_id)
+            if response["status"] == status:
+                results.append(job_id)
+        return results
+
+    def get_jobs_count_with_status(self, status):
+        """ return the number of jobs in the SQL Query server for the given `status`"""
+        jobs = self.get_jobs()
+        num_jobs = len(jobs[jobs["status"] == status])
+        return num_jobs
+
+    def get_number_running_jobs(self):
+        """ return the number of running jobs in the SQL Query server"""
+        return self.get_jobs_count_with_status('running')
+
+    def run_sql2(self, sql_stmt, pagesize=None, get_result=False, blocking=True):
```

Comment:
Try to find a better method name that reveals what it does differently than. run_sql.

Reply:
Any suggestions about the name?

In the run_sql() which is the blocking version, i.e. wait until the result is completed, it always returns the result which is not always the desired behavior, as the result can be too large, and the user doesn't want it to return back. The bool option `get_result` serves that purpose. So, rather than returning the result, it returns a namedtuple, which returns also the job_id so that the data can be pulled later if needed. 

NOTE: result.data is the one returned by `run_sql`, while result.job_id is the extra part.

Reply:
I understand and it would be better to rename run_sql to run_query or so. But we can't break current clients. So we need to find a different name for run_sql2 that reveals the fact that it doesn't return the data. Maybe execute_sql? You should also take away the blocking parameter here. As said above, setting max_tries in the client is all what is needed.

Reply:
Sure, I removed the blocking option. I renamed to `run_sql_v2`, as this behaves like a version of `run_sql` combining with `submit_sql`. 

`So we need to find a different name for run_sql2 that reveals the fact that it doesn't return the data. Maybe execute_sql?`

Just to clarify, run_sql2() does have the option to return data (when get_result=True) - it returns a namedtuple with 2 fields, (the field .data contain either returned data or None; the field .job_id holds the job_id). The online documentation should have all information a user needs. 

If you insist on using the name `execute_sql`, please let me know, I will change it accordingly.

Reply:
Yes, anything is better than a <method>_v2.

Reply:
Done in commit `0a974ee430a2206632c778a8254c5218f97694a4` 

I renamed to `execute_sql`.